src/wordlist.h

```diff
@@ -29,7 +28,7 @@ class WordlistIterator
     auto operator ==(const WordlistIterator &rhs) const { return this->w == rhs.w; }
     auto operator !=(const WordlistIterator &rhs) const { return this->w != rhs.w; }
 
-    inline reference operator *() const;
+    inline value_type operator *() const;
```

Comment:
FTR: We were forced to return value_type here (and, hence, remove std::input_iterator_tag iterator_category) because of unexplained build [failure](https://build.squid-cache.org/job/5-pr-test/7163/COMPILER=gcc,OS=centos-stream-8,label=pr-test-builder/console) with GCC on centos-stream-8:

```
20:02:14 ../../../../src/wordlist.h: In member function 'const char* const& WordlistIterator::operator*() const':
20:02:14 ../../../../src/wordlist.h:104:15: error: returning reference to temporary [-Werror=return-local-addr]
20:02:14      return w->key;
```